subsys/bluetooth/mesh/sensor_srv.c

```diff
@@ -487,12 +492,11 @@ static void cadence_set(struct bt_mesh_model *mod, struct bt_mesh_msg_ctx *ctx,
 		return;
 	}
 
-	model_send(mod, NULL, &rsp);
-
-respond:
 	if (ack) {
 		bt_mesh_model_send(mod, ctx, &rsp, NULL, NULL);
 	}
+
+	model_send(mod, NULL, &rsp);
```

Comment:
This will fail, as the bt_mesh_model_send callback above will corrupt the `rsp` buffer. 

`model_send` with `ctx=NULL` does not have this side effect, as it simply copies the buffer to the publish buffer without changing it. Either copy `rsp` to the publish buffer before calling bt_mesh_model_send, and use bt_mesh_model_send to do the publish as well, or create a second buffer with the same content. Same below

Reply:
Yes, you're absolutely right here. Sorry that I hadn't noticed it myself.